web/middleware/auth_ldap.py

```diff
@@ -0,0 +1,97 @@
+from web.settings import Settings as settings
+import asyncio
+import logging
+import sys
+import datetime
+import time
+import base64
+from sanic.exceptions import abort
+import ldap
+logger = logging.getLogger()
+
+async def get_ldap_connection(username, password):
+    ldap_object = ldap.initialize(settings.app['service']['ldap']['url'])
+    ldap_object.set_option(ldap.OPT_TIMEOUT, 8)
+    ldap_object.protocol_version = ldap.VERSION3
+    try:
+        ldap_object.simple_bind_s(username, password)
+        return ldap_object
+    except Exception as ex:
+        print("exception occured: {}".format(ex))
+        return None
+    return None
+
+def auth_user(username, password):
+    pass
+
+async def parse_auth(request):
+    auth_string = request.headers['authorization']
+    if auth_string.find("Basic ") == 0:
+        hash_string=auth_string[6:]
+        base64_bytes = hash_string.encode('ascii')
+        message_bytes = base64.b64decode(base64_bytes)
+        message = message_bytes.decode('ascii')
+        mess_arr = message.split(':')
```

Comment:
Hi Ivo,

please restrict the split here to happen just once, by using message.split(':', 1)

Reply:
Addressed